machine_learning/benchmarks/opflow_qnn_classifier_benchmark.py

```diff
@@ -28,21 +30,59 @@ class OpflowQnnClassifierBenchmarks(BaseClassifierBenchmark):
 
     version = 1
     timeout = 1200.0
-    params = [["dataset_1"], ["qasm_simulator", "statevector_simulator"]]
-    param_names = ["backend name"]
+    params = [
+        ["dataset_synthetic_classification", "dataset_iris"],
+        ["qasm_simulator", "statevector_simulator"],
+    ]
+    param_names = ["dataset", "backend name"]
+
+    def setup_dataset_synthetic_classification(self, X, y, quantum_instance_name):
+        """Training TwoLayerQNN for synthetic classification dataset."""
+
+        num_inputs = len(X[0])
+        self.y = 2 * y - 1  # in {-1, +1}
 
-    def setup(self, dataset, quantum_instance_name):
-        """setup"""
-        self.X = self.datasets[dataset]
-        num_inputs = len(self.X[0])
-        y01 = 1 * (np.sum(self.X, axis=1) >= 0)  # in { 0,  1}
-        self.y = 2 * y01 - 1  # in {-1, +1}
         opflow_qnn = TwoLayerQNN(num_inputs, quantum_instance=self.backends[quantum_instance_name])
         opflow_qnn.forward(self.X[0, :], np.random.rand(opflow_qnn.num_weights))
 
         self.opflow_classifier_fitted = NeuralNetworkClassifier(opflow_qnn, optimizer=COBYLA())
         self.opflow_classifier_fitted.fit(self.X, self.y)
 
+    def setup_dataset_iris(self, X, y, quantum_instance_name):
+        """Training TwoLayerQNN for iris classification dataset."""
+
+        num_inputs = len(X[0])
+
+        # keeping only two classes as TwoLayerQNN only supports binary classification
+        idx_binary_class = np.where(y != 2)[0]
+        self.X = X[idx_binary_class]
+        self.y = y[idx_binary_class]
+
+        feature_map = ZZFeatureMap(num_inputs)
+        ansatz = RealAmplitudes(num_inputs)
+
+        opflow_qnn = TwoLayerQNN(
+            num_inputs,
+            feature_map=feature_map,
+            ansatz=ansatz,
+            quantum_instance=self.backends[quantum_instance_name],
+        )
+        opflow_qnn.forward(self.X[0, :], np.random.rand(opflow_qnn.num_weights))
```

Comment:
Not quite sure why you need this forward pass.

Reply:
A forward pass was done for training the original synthetic dataset before I added iris, I just copied it over. I can remove it if you think it is unnecessary.

Reply:
After running a multiple times, it looks like any difference between runs with and without forward pass are unnoticeable in terms of score and time. I will remove this